src/topology/config/mod.rs

```diff
@@ -61,16 +61,16 @@ impl GlobalOptions {
     ) -> Result<PathBuf, crate::Error> {
         let data_dir = local_data_dir
             .or(self.data_dir.as_ref())
-            .ok_or_else(|| crate::box_error(DataDirError::MissingDataDir))?
+            .ok_or_else(|| Box::new(DataDirError::MissingDataDir) as crate::Error)? // Rust can't infer type here
```

Comment:
This is because Try is implemented for From not Into. To get around this you can

```rust
.ok_or_else(|| DataDirError::MissingDataDir)
.map_err(Into::into)?
```

I believe that should work.

Reply:
You can try `map_err(Box::new)` I think that should work too